stubs/inifile/inifile.pyi

```diff
@@ -99,7 +100,10 @@ class IniData(MutableMapping[str, str]):
     def __delitem__(self, name: str) -> None: ...
 
 class IniFile(IniData):
-    def __init__(self, filename: StrOrBytesPath | int, encoding: str | None = ..., dialect: Dialect | None = ...) -> None: ...
+    def __init__(self, filename: StrPath, encoding: str | None = ..., dialect: Dialect | None = ...) -> None: ...
+    filename: str
```

Comment:
Shouldn't this be the same type as the `__init__` parameter?

```suggestion
    filename: StrPath
```

Reply:
@AlexWaygood Maybe.  `Filename` does get coerced to a `str` [here](https://github.com/mitsuhiko/python-inifile/blob/dbdf3f319e6ef76dfa70b536defe916286a84f78/inifile.py#L500).

That brings up a somewhat related question:

Probably, `IniFile.filename` (as well as most or all of the other class attributes annotated in this module) should be treated as read-only.  To reflect that, I tried adding them as `@property`s, but `stubtest` didn't like that.
What's the best practice in this case?

Reply:
> @AlexWaygood Maybe. `Filename` does get coerced to a `str` [here](https://github.com/mitsuhiko/python-inifile/blob/dbdf3f319e6ef76dfa70b536defe916286a84f78/inifile.py#L500).

Oh, you're quite right! Sorry, I misread some code somewhere.

> To reflect that, I tried adding them as `@property`s, but `stubtest` didn't like that.
> What's the best practice in this case?

If necessary, we can just add allowlist entries to tell stubtest to be quiet about these "white lies" :) In general, we try to keep allowlist entries to a minimum, but if there's a good reason to tell stubtest to shut up in a certain situation, it's an option.

For an example of a third-party stubs package in typeshed with an allowlist, see https://github.com/python/typeshed/blob/main/stubs/colorama/%40tests/stubtest_allowlist.txt